### PR TITLE
Consistent body and header width

### LINKED ISSUE #480: Fix Table of Contents margin
The margin on the TOC page is not consistent with other pages. The main section of the page should always be as wide as the header.

image

### PR DESCRIPTION
**Pull request by catalinred**
This fixes #480

- The body is always as wide as the header due to the new `.container`class
- Within the contributors page, I've moved the counter to the left, as specified [here](https://github.com/HTTPArchive/almanac.httparchive.org/issues/548#issuecomment-559602998). Also, due to the new consistent `.container`, the contributors page is now supporting 5 columns of cards.

### CODE REVIEW

**Comment by catalinred** [comment]
@bazzadp I did a rebase and `git push -f` but still wasn't able to get rid of those extra old commits already merged. I guess I'm missing something.  :sigh:

**Comment by bazzadp** [comment]
Well at least you tried! 😀

Never worry for this PR. Will review as is.

Here’s what I normally do btw: https://stackoverflow.com/questions/7929369/how-to-rebase-local-branch-with-remote-master/18442755#18442755

But I’m no git expert myself to be honest.

**Comment by catalinred** [comment]
I'm working on #556 as well and would like to fix that too with this PR, if you agree!

**Comment by catalinred** [comment]
> Well at least you tried! 😀
> 
> Never worry for this PR. Will review as is.
> 
> Here’s what I normally do btw: https://stackoverflow.com/questions/7929369/how-to-rebase-local-branch-with-remote-master/18442755#18442755
> 
> But I’m no git expert myself to be honest.

Thank you for the link. Actually I am/supposed to be very familiar with `pull --rebase` but...

File: src/templates/fr/2019/contributors.html [hunk 1]
```diff
@@ -299,9 +294,11 @@ <h1 class="title title-lg">
       <img src="/static/images/avatars/9.jpg" height="100" width="100" loading="lazy" />
   </div>
   <div class="contributors {{ ' '.join(config.teams.keys()) }}">
-    <div class="contributors-count"><span id="filtered-contributors">{{ config.contributors.items() | length }}</span> personnes</div>
     <div class="filter-container">
-      <p>Filtrer par équipe&nbsp;:</p>
+      <p>
+        Filtrer par équipe&nbsp;:
```
Comment (bazzadp) [inline src/templates/fr/2019/contributors.html]: Any reason for this `&nbsp;` @borisschapira ?

File: src/static/css/2019.css [hunk 2]
```diff
@@ -30,6 +30,14 @@ a:focus, a:hover {
   text-decoration: underline;
 }
 
+/* The main wrapper, center aligned */
+.container {
+  max-width: 1280px;
```
Comment (bazzadp) [inline src/static/css/2019.css]: Why 1280 when previously it was 1324? Means all the chapters are a little bit thinner and longer.

```suggestion
  max-width: 1324px;
```

File: src/static/css/2019.css [hunk 3]
```diff
@@ -30,6 +30,14 @@ a:focus, a:hover {
...
+  padding-left: 20px;
+  padding-right: 20px;
```
Comment (bazzadp) [inline src/static/css/2019.css]: Any reason we need this padding? Wasn't there previously.

File: src/templates/en/2019/base.html [hunk 4]
```diff
@@ -73,7 +75,9 @@ <h2>
   </header>
   {% endblock %}
 
-  {% block main %}{% endblock %}
+  <div class="container">
+    {% block main %}{% endblock %}
+  </div>
 
   {% block footer %}
     <footer class="alt-bg">
```
Comment (bazzadp) [inline src/templates/en/2019/base.html]: Why is the footer not in a container div? Currently it's very wide compared to everything else on the page.

File: src/static/css/index.css [hunk 5]
```diff
@@ -17,6 +17,19 @@ header.alt-bg {
   background-color: transparent;
 }
 
+header .container {
+  max-width: none;
```
Comment (bazzadp) [inline src/static/css/index.css]: Why no max-width? I agree that is the current design but seems weird that Home page is very wide, but rest are constrained. Maybe not part of this PR and maybe one to raise in an issue and discuss, but looks weird to me none-the-less so thought I'd ask.

File: src/static/css/index.css [hunk 6]
```diff
@@ -17,6 +17,19 @@ header.alt-bg {
...
+  padding-left: 60px;
+  padding-right: 60px;
+}
+
+@media (max-width: 600px) {
+  header .container {
+    padding-left: 30px;
+    padding-right: 30px;
```
Comment (bazzadp) [inline src/static/css/index.css]: Why 30px instead of 20px? It makes the menu appear weird on mobile for index page only:

image

File: src/static/css/2019.css (hunk 2 above)
Comment (catalinred) [reply src/static/css/2019.css]: As I wrote [here](https://github.com/HTTPArchive/almanac.httparchive.org/issues/480#issuecomment-556009375), the Methodology page was relying on a `max-width: 1324px` while Table of Contents & Contributors pages on `max-width: 1024px`. 

While I thought 1324px is an arbitrary value chosen for a breakpoint and 1024px way too small, considering a 1200px is recommended value for desktop-and-up displays when it comes to how [frameworks structure their breakpoints](https://getbootstrap.com/docs/4.3/layout/overview/#responsive-breakpoints) I decided to go with a median to say so between the 1200 which I well guessed you will consider being too low and the actual 1324.

I'm happy to update this to 1324 if that's what you want but I still consider this value is "hardcoded" and too big.

File: src/static/css/2019.css (hunk 3 above)
Comment (catalinred) [reply src/static/css/2019.css]: When defining a container element, you want to set left and right padding "bleeds" which are both visible and effective on mobile displays. 

Also, I could have used a shorthand like `padding: 0 20px`to get rid of an extra line but this could have caused issues when combining .e.g. with other padding helpers - helpers which we don't have now but we might in the future.

File: src/templates/fr/2019/contributors.html (hunk 1 above)
Comment (catalinred) [reply src/templates/fr/2019/contributors.html]: Noticed that too, wanted to change but as this got to master, I thought there might be a reason for that.

File: src/static/css/2019.css (hunk 2 above)
Comment (bazzadp) [reply src/static/css/2019.css]: OK I hadn't considered the chapters changing. I'm OK with 1200.

File: src/static/css/2019.css (hunk 3 above)
Comment (bazzadp) [reply src/static/css/2019.css]: Then is it needed on desktop? Just with the reduction from 1324 -> 1200 and then losing another 40px here, we've effectively gone from 1324 to 1160. Not that that is necessarily bad, but thought I'd ask the question.

File: src/templates/fr/2019/contributors.html (hunk 1 above)
Comment (bazzadp) [reply src/templates/fr/2019/contributors.html]: I saw remove. Unless @borisschapira gives good reason to keep it.

Comment (borisschapira) [reply src/templates/fr/2019/contributors.html]: French microtypography. We have non-break spaces before ":", ";", "!", "?", between numbers and units ("14&nbsp;%")…

File: src/templates/en/2019/base.html (hunk 4 above)
Comment (catalinred) [reply src/templates/en/2019/base.html]: I decided to let the footer that wide on purpose, as I thought it looks pretty good to me - from a design perspective - considering its current content: logo, menu, social, icons, copyright info.

Also, having a full-width footer is most of the time a common practice in web design/development.

File: src/templates/fr/2019/contributors.html (hunk 1 above)
Comment (bazzadp) [reply src/templates/fr/2019/contributors.html]: OK. As long as there's a reason. Resolving this with no change then.

File: src/templates/en/2019/base.html (hunk 4 above)
Comment (bazzadp) [reply src/templates/en/2019/base.html]: Yeah seen it before but not a fan of that design principal. Especially when not a huge amount in our footer to need that. But hey that's a personal opinion and sure there's plenty of others who prefer this design.

The [original design](https://github.com/HTTPArchive/almanac.httparchive.org/issues/46#issuecomment-515966879) has narrower footer, then again it has wider header - that we are changing with this PR - so not sure that's helping my argument! 😀

Happy to resolve this with no change if you think it's working as intended.

File: src/static/css/index.css (hunk 5 above)
Comment (catalinred) [reply src/static/css/index.css]: Sure, that's a good question to ask.

I didn't ask about the current design decisions either, as I wasn't involved from the beginning and that wouldn't have been fair coming from me. 

It would be a good idea to raise this as part of a different issue and besides that, we should take into account the fact that the header "jumps" on the index due to the "Start exploring" right upper button. I'm pretty sure that wasn't on purpose and we should address that too maybe.

Comment (bazzadp) [reply src/static/css/index.css]: Raised #574 . Let's resolve this for this PR.

File: src/static/css/index.css (hunk 6 above)
Comment (catalinred) [reply src/static/css/index.css]: That is correct, thanks for the heads up. 

`20` is the value I was aiming at, see the existing comment for the responsive nav: `/* overcome the parent container's 20px side padding */  `. I guess I got pretty tired at the end. Fixed it.

File: src/templates/en/2019/base.html (hunk 4 above)
Comment (catalinred) [reply src/templates/en/2019/base.html]: TBH lots of things are a bit different compared to the original design, and that's not necessarily a bad thing, because this is what it happens usually when you implement a design :)

Yes, in my opinion, the footer looks and works as intended. Thanks!

File: src/static/css/2019.css (hunk 3 above)
Comment (catalinred) [reply src/static/css/2019.css]: Having set 1280px for the `.container`, the total "remaining" available space for the content is 1240px and not 1160px.

**Comment by bazzadp** [comment]
@HTTPArchive/developers anyone free to review this change? If not will merge this PR this weekend.
